oceanstream/L3_regridded_data/applying_masks_handler.py

```diff
@@ -0,0 +1,171 @@
+"""
+    applying_masks_handler.py
+    -------------------------
+    Description: This module provides functionalities for applying a sequence of
+    selected masks and background noise removal to echosounder datasets based on
+    specific methodologies.
+
+    Masks/Processes supported in this module include:
+    **Noise-related masks/processes, namely:**
+    - 'mask_false_seabed'
+    - 'mask_seabed'
+    - 'mask_impulse'
+    - 'mask_attenuation'
+    - 'mask_transient'
+    - 'remove_background_noise'
+
+    **Organism-related masks, namely:**
+    - 'mask_krill'
+    - 'mask_gas_bearing_organisms'
+    - 'mask_fluid_like_organisms'
+    - 'mask_shoal'
+
+    Note: If other masks that aren't supported by this module's functions are
+    attached as variables to the dataset, they can be applied using
+    `echopype.mask.apply_mask`. For details, consult the echopype documentation.
+
+    Key Functions:
+    - `apply_selected_noise_masks_and_or_noise_removal`: For applying selected
+    noise masks and/or noise removal.
+
+    - `apply_mask_organisms_in_order`: For applying organism-related masks in
+    a user-defined order.
+"""
+
+
+import pathlib
+from typing import Union
+
+import echopype as ep
+import xarray as xr
+
+from oceanstream.L2_calibrated_data import background_noise_remover
+
+
+def apply_selected_noise_masks_and_or_noise_removal(
+    ds: Union[xr.Dataset, str, pathlib.Path],
+    processes_to_apply: dict,
+) -> xr.Dataset:
+    """
+    Apply a sequence of selected masks and background noise removal (if provided in the processes_to_apply)
+    to the given dataset based on the specifications in the article:
+    > Haris, K., et al. "Sounding out life in the deep using acoustic data from ships of opportunity."
+    > Scientific Data 8.1 (2021): 23.
+
+    Order of cleaning process:
+    1. False seabed mask (if provided)
+    2. Seabed mask (if provided)
```

Comment:
These should be applied last, I think

Reply:
The original paper describing the background noise removal, https://academic.oup.com/icesjms/article/64/6/1282/616894 mentions quite a bit of sensitivity in the algorithm, based on how much data depth is included. From what I remember, the false seabed mask removes quite a bit of data below the echo itself. Also, if you look at figure 8 here, you can see the algorithm being applied on a dataset that includes a false seabed echo